app/models/setting.rb

```diff
@@ -83,4 +84,24 @@ def self.invoice_reminder_in_days
   def self.wishlist_size
     Setting.find_by(code: :wishlist_size).value.to_i
   end
+
+  def self.violations_count_regulations_link
+    hash = Setting.find_by(code: :violations_count_regulations_link)&.value
+    hash.present? ? JSON.parse(hash).with_indifferent_access[I18n.locale] : nil
+  end
+
+  private
+
+  def valid_violations_count_regulations_link_format
+    return unless code == 'violations_count_regulations_link'
+    return if valid_violations_count_json?(value)
+
+    errors.add(:value, :invalid)
+  end
+
+  def valid_violations_count_json?(value)
+    JSON.parse(value).is_a?(Hash)
+  rescue JSON::ParserError
```

Comment:
Avoid failing silently.

Reply:
Hmmm, maybe I should save  exception message in validation error?
I'll do so anyway.

Reply:
Do not catch without a serious reason, or it will lead to hard-to-find errors and long debugging sessions. Let it fail. In this case I would rethink  the design. The main question I had, was "Why should it fail while parsing?". If it's totally unexpected, let it fail. If it is expected, something is wrong with the design, in my opinion.